iothub/device/src/ClientFactory.cs

```diff
@@ -113,7 +114,7 @@ internal static InternalClient Create(string hostname, IAuthenticationMethod aut
             var connectionStringBuilder = IotHubConnectionStringBuilder.Create(hostname, gatewayHostname, authenticationMethod);
 
             // Make sure client options is initialized with the correct transport setting.
-            EnsureOptionsIsSetup(connectionStringBuilder.Certificate, ref options);
+            EnsureOptionsIsSetup(connectionStringBuilder.Certificate, null, ref options);
```

Comment:
Adds support for copying the proxy too, but this factory method doesn't have a ClientOptions.